install/_test_setup.sh

```diff
@@ -0,0 +1,50 @@
+set -euo pipefail
```

Comment:
Not following this file but the rest looks pretty good. Should we add this to our CI suite?

Reply:
This file is the test framework, basically. `setup` is the important part, it clones the repo into a temp directory, propagates any local changes (expected to be a no-op in CI, but quite useful in local dev: since we symlink changed files, you can edit files in your main working copy and re-run the test inside of the sandbox), and wires up cleanup on teardown (unless `DEBUG` envvar is set).

Reply:
It is not clear how one would use this to test, that's what I was thinking.

Maybe if we can add this to CI, we can get cues from there.

Reply:
Ah. For that, check out `relay-credentials-test.sh` on this PR, more tests on the next ... :)